src/components/screens/SwappingScreen.tsx

```diff
@@ -130,17 +113,24 @@ const SwappingScreen = () => {
           </div>
         </div>
 
-        {txStatus === undefined ? (
-          <span className="text-white text-center mt-8 text-sm px-4">Awaiting approval from your wallet...</span>
-        ) : null}
-
         <div className="flex flex-col w-full justify-center items-center px-5 mt-7">
-          {swapState === 'loading' && (
+          {(txStatus?.status === 'loading' ||
+            txStatus?.status === 'pending-approval' ||
+            txStatus?.status === 'sending') && (
```

Comment:
Just do a variable `isLoading` to check for these three status, so that here and any place require isLoading check can use it